### PR TITLE
feat(license): Update Whitelist for License

### PR DESCRIPTION
**Pull request by tuannn2**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

### Feature
- Feature Update Whitelist in License Page

### How To Test?
- Run command
```
curl 'https://sw360.org/api/licenses/Apache-2.0/whitelist' -i -X PATCH \
    -H 'Content-Type: application/hal+json' \
    -H 'Authorization: BearereyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsic3czNjAtUkVTVC1BUEkiXSwidXNlcl9uYW1lIjoiYWRtaW5Ac3czNjAub3JnIiwic2NvcGUiOlsiYWxsIl0sImV4cCI6MTcwNDg4MzY5OSwiYXV0aG9yaXRpZXMiOlsiUkVBRCIsIldSSVRFIl0sImp0aSI6ImRJc0ZOSml3ckhHRjE1ZHJrZzBCX1JSX21rVSIsImNsaWVudF9pZCI6InRydXN0ZWQtc3czNjAtY2xpZW50In0.iuADty2_ZFsM1ETIqAJIV43oM9djLO1aNnNfYyArigg71mbFqjsrxL0q_n9GZcCqtLIJQfIiIjv6g7v8SFC-fS5RsIw0rDRFjHAydIUWdr8lhG96YomWAFfJNicIng6h1q6FU-JUbzPgneYStkGJQ1RXUWUMGO9_nMzimED7oI56JPL4DiicyS0Q7QzNnvTIx3zyub7SjWXGEM82Lc4DzweLIdL3G7tmklHxU2nIbL0ch9xhhEjgZiiTik3t0s-z-LXVlLqCui7lgvHi1n9CZ6VKdyA_s5KtjrdLSBYlWa2R8l6_dyPY5lvVhyIUr6ozx5mCHUPpfVsf1uc0Wedcsg' \
    -H 'Accept: application/hal+json' \
    -d '{
  "0002" : true,
  "0001" : true
}'
```

- Output 
```
{
  "externalLicenseLink" : "https://spdx.org/licenses/Apache-2.0.html",
  "note" : "License's Note",
  "externalIds" : {
    "Trove" : "License :: OSI Approved :: Apache Software License",
    "SPDX" : "Apache-2.0"
  },
  "additionalData" : {
    "Key" : "Value"
  },
  "OSIApproved" : "NA",
  "FSFLibre" : "NA",
  "obligations" : [ {
    "type" : "obligation",
    "text" : "This is text of Obligation 1",
    "whitelist" : [ "Department" ],
    "title" : "Obligation 1",
    "obligationLevel" : "LICENSE_OBLIGATION",
    "obligationType" : "PERMISSION"
  }, {
    "type" : "obligation",
    "text" : "This is text of Obligation 2",
    "whitelist" : [ "Department2" ],
    "title" : "Obligation 2",
    "obligationLevel" : "LICENSE_OBLIGATION",
    "obligationType" : "OBLIGATION"
  } ],
  "obligationDatabaseIds" : [ "0002", "0001" ],
  "text" : "placeholder for the Apache 2.0 license text",
  "checked" : true,
  "shortName" : "Apache 2.0",
  "fullName" : "Apache License 2.0",
  "_links" : {
    "self" : {
      "href" : "https://sw360.org/api/licenses/Apache-2.0"
    }
  },
  "_embedded" : {
    "sw360:obligations" : [ {
      "text" : "This is text of Obligation 1",
      "whitelist" : [ "Department" ],
      "title" : "Obligation 1",
      "obligationType" : "PERMISSION",
      "_links" : {
        "self" : {
          "href" : "https://sw360.org/api/obligations/0001"
        }
      }
    }, {
      "text" : "This is text of Obligation 2",
      "whitelist" : [ "Department2" ],
      "title" : "Obligation 2",
      "obligationType" : "OBLIGATION",
      "_links" : {
        "self" : {
          "href" : "https://sw360.org/api/obligations/0002"
        }
      }
    } ]
  }
}
```

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by nikkuma7** [comment]
Hi @tuannn2 ,
I have created the obligation licenses and linked with the licenses in license tab. As we can see in the below SS.
image

I have linked the obligation licenses with OBSD license.
image

Getting "No license details found in the database" Even i have the license with the same name in the database.
image

Please let me know if any configuration or changes required.

**Comment by tuannn2** [comment]
> Getting "No license details found in the database" Even i have the license with the same name in the database. image
> 
> Please let me know if any configuration or changes required.

**Comment by tuannn2** [comment]
> Hi @tuannn2 , I have created the obligation licenses and linked with the licenses in license tab. As we can see in the below SS. image
> 
> I have linked the obligation licenses with OBSD license. image
> 
> Getting "No license details found in the database" Even i have the license with the same name in the database. image
> 
> Please let me know if any configuration or changes required.

Hi @nikkuma7,
You can try it with this url: ```https://localhost:8080/api/licenses/OBSD/whitelist``` 
Instead of 
FeedbackTestWhitelist

**Comment by nikkuma7** [comment]
getting error : status code (501) server does not support the functionality required to fulfill the request.
image

**Comment by tuannn2** [comment]
> getting error : status code (501) server does not support the functionality required to fulfill the request. image

Did you use OSBD or 0SBD? 
Please copy the```License Shortname``` of license.

**Comment by nikkuma7** [comment]
Test successful.
image

File: rest/resource-server/src/docs/asciidoc/licenses.adoc [hunk 1]
```diff
@@ -96,6 +96,20 @@ include::{snippets}/should_document_update_license/http-response.adoc[]
 ===== Links
 include::{snippets}/should_document_update_license/links.adoc[]
 
+[[resources-update-whitelist-license]]
+==== Update whitelist a license
```
Comment (rudra-superrr) [inline rest/resource-server/src/docs/asciidoc/licenses.adoc:100]: can you change the heading to, Update whitelist for license's obligations for better understanding.

Comment (hoangnt2) [reply rest/resource-server/src/docs/asciidoc/licenses.adoc]: @rudra-superrr  Thanks for your comment. I have fixed it. Please help me to review the changes

**Comment by rudra-superrr** [comment]
LGTM.

**Pull request merged**